lib/logstash/config/mixin.rb

```diff
@@ -85,6 +85,11 @@ def config_init(params)
       end
     end
 
+    params.reject! do |name, value| 
+      opts = self.class.get_config[name]
+      opts.include?(:deprecated)
```

Comment:
This aint gonna work. Check for `obsolete` happens before `params.reject!` is executed. This will bail once params with `obsolete` is encountered before this line. 

I think the correct issue (or place to fix this) is when you do `subject.params`, default params are created. This includes the obsolete config options with default values.